portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+/** 
+ * FUNCTIONS 
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/* NAV */
+const mainViews = document.getElementsByClassName('main-view');
+const dropdownMenu = document.getElementById('dropdown-menu');
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+// hide dropdown menu and show main view elements
+const hideMenuShowMain = () => {
+  dropdownMenu.style.display = 'none';
+  for (let mainView of mainViews) {
+    mainView.style.display = 'flex';
+  }
+}
+
+ /**
+  * ADD EVENT LISTENERS
+  */
+
+/* NAV */
+const menuIcon = document.getElementById('menu-icon');
+// hide main view elements and display dropdown menu
+menuIcon.onclick = () => {
+  for (let mainView of mainViews) {
+   mainView.style.display = 'none';
+  }
+  dropdownMenu.style.display = 'block';
```

Comment:
what if dropwdownMenu is null?

Reply:
Oh I have to check that too? I didn't make program it defensively because I am the only one working on this. 😅 Will modify it though!

Reply:
Yeah I would say we should add that. Right now things are all identified by "strings", and there's not a single source of truth. E.g, one day you change html ids and forget to change the id in this js file, nothing will start complaining errors until the website actually runs.

Reply:
This is true! Thanks :D

Reply:
I have included error handling. What do you think about it? Honestly, I am not very familiar with error handling in JS. Thanks in advance! :D

Reply:
No worries, I guess this is more of an issue with this vanila html/css, which I'm also very familiar of some of the best practices.

Reply:
Cool! Do you have any advice as to how I can pick up these practices? :D